openapi-codegen/core/src/main/scala/sttp/tapir/codegen/ClassDefinitionGenerator.scala

```diff
@@ -176,6 +180,35 @@ class ClassDefinitionGenerator {
     if (optional || !required) s"Option[$tpe]" else tpe
   }
 
+  private def renderDefault(required: Boolean, default: RenderableValue, schemaType: OpenapiSchemaType): String = {
+    val rendered = default.render
+    val base = (default, schemaType) match {
+      case (ReifiableValueLong(_), OpenapiSchemaInt(_))        => rendered.stripSuffix("L")
+      case (ReifiableValueLong(_), OpenapiSchemaFloat(_))      => rendered.stripSuffix("L") + "f"
+      case (ReifiableValueLong(_), OpenapiSchemaDouble(_))     => rendered.stripSuffix("L") + "d"
+      case (ReifiableValueDouble(_), OpenapiSchemaFloat(_))    => rendered.stripSuffix("d") + "f"
+      case (ReifiableValueString(_), OpenapiSchemaDate(_))     => throw new NotImplementedError("Default date fields are not supported")
+      case (ReifiableValueString(_), OpenapiSchemaDateTime(_)) => s"java.time.Instant.parse(${rendered})"
+      case (ReifiableValueString(_), OpenapiSchemaByte(_))     => throw new NotImplementedError("Default byte fields are not supported")
+      case (ReifiableValueString(_), OpenapiSchemaBinary(_))   => s"""$rendered.getBytes("utf-8")"""
+      case (ReifiableValueString(_), OpenapiSchemaUUID(_))     => s"java.util.UUID.fromString(${rendered})"
+      // Nasty hacks so that enum defaults sometimes work
```

Comment:
**question:** Do you know if there are cases for which it wouldn't work?

Reply:
So defaults values that are themselves case classes won't work unless the specified fields are all precisely the guessed types from the declaration, because at this point we don't know what the ref resolves to, so don't know if a field is really a string or an enum for example. Fixing that would be tricky because we'd need to perform the structural lookups and fold the types over the model. I could have a crack at addressing that, but I don't know how successful I'd be...